guardian/riff-raff
# Improve the reporting of ASG discovery

## What does this change?
This improves the way that RiffRaff displays tasks that act on AWS autoscaling groups in two ways:
 - the tasks now contain the related ASG name
 - the verbose discovery logic has been improved to exhaustively list the tag requirements

Some housekeeping was done to:
 - switch the ASG helpers to use the paginators available in SDKv2 rather than implementing our own custom recursive function
 - refactor the lookup code to take generic tag requirements and move the ASG specific logic mapping into the AutoScaling deployment class
 - remove the unused and probably dangerous `elasticsearch` deployment type

https://trello.com/c/oj4zjP28/853-improve-asg-identification-in-riffraffs-autoscaling-deploy-type

## How to test
Run a deploy in CODE and verify that there is now an ASG name on relevant tasks and more information in the logs.

Before:
<img width="575" alt="image" src="https://user-images.githubusercontent.com/1236466/134168536-4d85312b-8fb0-4c2d-a753-ab2dd6be360a.png">

After:
<img width="575" alt="image" src="https://user-images.githubusercontent.com/1236466/134168639-4e271fff-51fb-48f7-9abe-8ba947707cf1.png">


## How can we measure success?
Deploy logs will now contain much clearer information about which ASG was chosen and how it was chosen.
